src/OAuth2User.ts

```diff
@@ -39,13 +40,17 @@ export class OAuth2User implements OAuthClient {
   code_verifier?: string;
   code_challenge?: string;
   private _refreshAccessTokenPromise: Promise<{token: Token}> | null;
+  private _tokenEvents = new EventEmitter();
   constructor(options: OAuth2UserOptions) {
     const { token, ...defaultOptions } = options;
     this.options = {client_secret: '', ...defaultOptions};
     this.token = token;
     this._refreshAccessTokenPromise = null;
   }
-
+  
+  on(eventName: string, listener: (tokens: Token) => void) {
```

Comment:
can you type the eventName so it only accepts the ones we support - `eventName: "tokens"`